web/handlers/http.go

```diff
@@ -122,6 +123,42 @@ func New(
 			}
 		}),
 
+		render.InjectTemplateFunc("language_count", func(r *http.Request) interface{} {
+			return func() int {
+				return len(locHelper.ListLanguages())
+			}
+		}),
+
+		render.InjectTemplateFunc("list_languages", func(r *http.Request) interface{} {
+			csrfElement := csrf.TemplateField(r)
+
+			createFormElement := func(postRoute, tag, translation, classList string) string {
+				return fmt.Sprintf(`
```

Comment:
I'd invite you to try this as a standalone template. 6 format inputs is quite a lot, i think.

I'd place this above the `render.Options` (or somehwere sensible, this func is getting quite long..)


```go
type languageChangeTemplateData struct {
	Action    string
	CSRFField template.HTML
	Language  string
	Page      string
	FormValue string
	ClassList string
}

languageChangeForm, err := template.New("name").Parse(`
            <form
              action="{{.Action}}"
              method="POST"
              >
              {{.CSRFField}}
              <input type="hidden" name="lang" value="{{.Language}}">
              <input type="hidden" name="page" value="{{.Page}}">
              <input
                type="submit"
                value="{{.FormValue}}"
                class="{{.ClassList}}"
                />
            </form>
`)
if err!=nil { ... }
```

And then just call this inside the `list_languages` template func:

```go
	var (
		buf       bytes.Buffer
		languages = locHelper.ListLanguages()
	)

	for _, entry := range languages {

		data := languageChangeTemplateData{
			Action:    postRoute,
			CSRFField: csrfElement,
			Language:  tag,
			Page:      r.RequestURI,
			FormValue: translation,
			ClassList: classList,
		}
		err = languageChangeForm.Execute(&buf, data)
		if err != nil {
			return "", err
		}
	}

	return template.HTML(buf.String()), nil

```

ps: always writing into `buf` will also save us from a medium expensive/wastefull "small strings onto one big one".

Reply:
the `data := languageChangeTemplateData{ ... }` surely needs to be changed a little but I think you get the idea.

Reply:
wdyt about the implementation in 546b249? :)